tests/integration/test_charm.py

```diff
@@ -51,36 +46,65 @@ async def test_build_and_deploy(
             )
         )
         tg.create_task(
-            ops_test.model.deploy(
-                str(filesystem_client),
-                application_name=FILESYSTEM_CLIENT,
-                num_units=0,
+            build_and_deploy_charm(
+                ops_test, filesystem_client_charm, application_name=FILESYSTEM_CLIENT, num_units=0
             )
         )
-        tg.create_task(
-            ops_test.model.deploy(
-                str(nfs_server_proxy),
-                application_name=NFS_SERVER_PROXY,
-                config={"hostname": nfs_info.hostname, "path": nfs_info.path},
+
+        async def deploy_nfs_proxy():
+            """Deploy an NFS server and pass its info to a new `nfs_server_proxy_charm`."""
+
+            async def deploy_charm():
+                nfs_server_proxy = await nfs_server_proxy_charm
+                (
+                    await ops_test.model.deploy(
+                        str(nfs_server_proxy), base=charm_base, application_name=NFS_SERVER_PROXY
+                    ),
+                )
+
+            nfs_info, _ = await asyncio.gather(bootstrap_nfs_server(ops_test), deploy_charm())
```

Comment:
[comment]: I feel like this would be a great time for `async lambda` to be a thing...